File: zoomx/src/main/java/com/zoomx/zoomx/ui/info/InfoFragment.java
package com.zoomx.zoomx.ui.info;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.zoomx.zoomx.R;
import com.zoomx.zoomx.model.InfoModel;
import com.zoomx.zoomx.util.PhoneUtils;

import java.util.ArrayList;

/**
 * Created by Ahmed Fathallah on 2/17/2018.
 */

public class InfoFragment extends Fragment {

    private View view;

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        view = inflater.inflate(R.layout.info_fragment, container, false);
        initUI();
        return view;
    }

    private void initUI() {
        ((InfoActivity) getActivity()).getSupportActionBar().setTitle(R.string.info_screen_title);

        InfoSectionView appInfo = view.findViewById(R.id.info_app_view);
        appInfo.setInfoModels(createAppInfo());
        appInfo.setSectionTitle(getString(R.string.info_app_section_title));

        InfoSectionView deviceInfo = view.findViewById(R.id.info_device_view);
        deviceInfo.setInfoModels(createDeviceInfo());
        deviceInfo.setSectionTitle(getString(R.string.info_device_section_title));
    }

    private ArrayList<InfoModel> createDeviceInfo() {
        ArrayList<InfoModel> infoModels = new ArrayList<>();
        InfoModel model = new InfoModel(InfoModel.TEXT_VIEW_TYPE, getString(R.string.info_device_model), PhoneUtils.getDeviceModel());
        infoModels.add(model);
        model = new InfoModel(InfoModel.TEXT_VIEW_TYPE, getString(R.string.info_device_density), PhoneUtils.getDensity(getContext()));
        infoModels.add(model);
        model = new InfoModel(InfoModel.TEXT_VIEW_TYPE, getString(R.string.info_device_resolution), PhoneUtils.getDeviceResolution(getContext()));
        infoModels.add(model);
        model = new InfoModel(InfoModel.TEXT_VIEW_TYPE, getString(R.string.info_release), PhoneUtils.getAndroidRelease());
        infoModels.add(model);
        model = new InfoModel(InfoModel.TEXT_VIEW_TYPE, getString(R.string.info_api), String.valueOf(PhoneUtils.getAndroidApi()));
        infoModels.add(model);
        return infoModels;
    }

    private ArrayList<InfoModel> createAppInfo() {
        ArrayList<InfoModel> infoModels = new ArrayList<>();
        InfoModel model = new InfoModel(InfoModel.TEXT_VIEW_TYPE, getString(R.string.info_package_name), PhoneUtils.getPackageName(getContext()));
        infoModels.add(model);
        model = new InfoModel(InfoModel.TEXT_VIEW_TYPE, getString(R.string.info_version_name), PhoneUtils.getAppVersionName(getContext()));
        infoModels.add(model);
        model = new InfoModel(InfoModel.TEXT_VIEW_TYPE, getString(R.string.info_version_code), String.valueOf(PhoneUtils.getAppVersion(getContext())));
        infoModels.add(model);
        return infoModels;
    }
}

File: zoomx/src/main/java/com/zoomx/zoomx/ui/info/InfoFragment.kt
package com.zoomx.zoomx.ui.info

import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.zoomx.zoomx.R
import com.zoomx.zoomx.model.InfoModel
import com.zoomx.zoomx.util.PhoneUtils
import java.util.*

/**
 * Created by Ahmed Fathallah on 2/17/2018.
 */

class InfoFragment : Fragment() {

    private var view: View? = null

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        view = inflater!!.inflate(R.layout.info_fragment, container, false)
        initUI()
        return view
    }

    private fun initUI() {
        (activity as InfoActivity).supportActionBar!!.setTitle(R.string.info_screen_title)

        val appInfo = view!!.findViewById<InfoSectionView>(R.id.info_app_view)
        appInfo.setInfoModels(createAppInfo())
        appInfo.setSectionTitle(getString(R.string.info_app_section_title))

        val deviceInfo = view!!.findViewById<InfoSectionView>(R.id.info_device_view)
        deviceInfo.setInfoModels(createDeviceInfo())
        deviceInfo.setSectionTitle(getString(R.string.info_device_section_title))

        val actions = view!!.findViewById<InfoSectionView>(R.id.info_actions_view)
        actions.setInfoModels(createSettings())
        actions.setSectionTitle(getString(R.string.info_settings_section_title))
    }

    private fun createSettings(): ArrayList<InfoModel> {
        val infoModels = ArrayList<InfoModel>()
        var model = InfoModel(InfoModel.BUTTON_VIEW_TYPE, getString(R.string.info_clear_app_data), null!!, Runnable { PhoneUtils.clearAppData(activity) })
        infoModels.add(model)
        model = InfoModel(InfoModel.BUTTON_VIEW_TYPE, getString(R.string.info_uninstall), null!!, Runnable { PhoneUtils.uninstall(activity) })
        infoModels.add(model)
        return infoModels
    }


    private fun createDeviceInfo(): ArrayList<InfoModel> {
        val infoModels = ArrayList<InfoModel>()
        var model = InfoModel(InfoModel.TEXT_VIEW_TYPE, getString(R.string.info_device_model), PhoneUtils.deviceModel)
        infoModels.add(model)
        model = InfoModel(InfoModel.TEXT_VIEW_TYPE, getString(R.string.info_device_density), PhoneUtils.getDensity(context))
        infoModels.add(model)
        model = InfoModel(InfoModel.TEXT_VIEW_TYPE, getString(R.string.info_device_resolution), PhoneUtils.getDeviceResolution(context))
        infoModels.add(model)
        model = InfoModel(InfoModel.TEXT_VIEW_TYPE, getString(R.string.info_release), PhoneUtils.androidRelease)
        infoModels.add(model)
        model = InfoModel(InfoModel.TEXT_VIEW_TYPE, getString(R.string.info_api), PhoneUtils.androidApi.toString())
        infoModels.add(model)
        return infoModels
    }

    private fun createAppInfo(): ArrayList<InfoModel> {
        val infoModels = ArrayList<InfoModel>()
        var model = InfoModel(InfoModel.TEXT_VIEW_TYPE, getString(R.string.info_package_name), PhoneUtils.getPackageName(context))
        infoModels.add(model)
        model = InfoModel(InfoModel.TEXT_VIEW_TYPE, getString(R.string.info_version_name), PhoneUtils.getAppVersionName(context))
        infoModels.add(model)
        model = InfoModel(InfoModel.TEXT_VIEW_TYPE, getString(R.string.info_version_code), PhoneUtils.getAppVersion(context).toString())
        infoModels.add(model)
        return infoModels
    }
}
